src/shared/ui/navigation/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,201 @@
+"use client";
+
+import * as DropdownMenuPrimitive from "@radix-ui/react-dropdown-menu";
+import { HiChevronRight } from "react-icons/hi2";
+import { IoMdCheckmark } from "react-icons/io";
+import { RxDotFilled } from "react-icons/rx";
+
+import * as React from "react";
+
+import { twJoin } from "tailwind-merge";
+
+export const DropdownMenu = DropdownMenuPrimitive.Root;
+
+export const DropdownMenuTrigger = DropdownMenuPrimitive.Trigger;
+export const DropdownMenuGroup = DropdownMenuPrimitive.Group;
+
+export const DropdownMenuPortal = DropdownMenuPrimitive.Portal;
+
+export const DropdownMenuSub = DropdownMenuPrimitive.Sub;
+
+export const DropdownMenuRadioGroup = DropdownMenuPrimitive.RadioGroup;
+
+export const DropdownMenuSubTrigger = React.forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.SubTrigger>,
+  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubTrigger> & {
+    inset?: boolean;
+  }
+>(function DropdownMenuSubTrigger(
+  { className, inset, children, ...props },
+  ref,
+) {
+  return (
+    <DropdownMenuPrimitive.SubTrigger
+      ref={ref}
+      className={twJoin(
+        "focus:bg-accent data-[state=open]:bg-accent flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none",
+        inset && "pl-8",
+        className,
+      )}
+      {...props}
+    >
+      {children}
+      <HiChevronRight className="ml-auto h-4 w-4" />
+    </DropdownMenuPrimitive.SubTrigger>
+  );
+});
+
+export const DropdownMenuSubContent = React.forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.SubContent>,
+  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubContent>
+>(function DropdownMenuSubContent({ className, ...props }, ref) {
+  return (
+    <DropdownMenuPrimitive.SubContent
+      ref={ref}
+      className={twJoin(
+        "bg-popover text-popover-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 min-w-[8rem] overflow-hidden rounded-md border p-1 shadow-lg",
```

Comment:
I think we can remove these classes:
`bg-popover text-popover-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50`

1. `bg-popover text-popover-foreground` are classes we didn't defined in our tailwind config
2. The `data-...` won't work because we'd have to install a [tailwind plugin](https://github.com/jamiebuilds/tailwindcss-animate) first to make them work (we'll install it later if necessary).
3. `z-50` should not be required for now. When we'll need `z-index`, we'll use layers instead of arbitrary values for easier management :) (`z-popover`, `z-modal`, `z-toast`)
4. Can we replace `p-1` with `p-spacing-tighter-4` instead?

Reply:
YAGNI again.